src/Core/Snippets/Snippets.cs

```diff
@@ -149,6 +149,14 @@ public Snippet Compile(string code)
 
             try
             {
+                // strip any @EntryPoint() attributes from the snippet
+                var entryPointPattern = @"@\s*EntryPoint\s*\(\s*\)";
+                if (Regex.IsMatch(code, entryPointPattern))
+                {
+                    code = Regex.Replace(code, entryPointPattern, string.Empty);
+                    logger.LogWarning("@EntryPoint()", "@EntryPoint() attributes are ignored when compiling Q# code in this workspace.");
```

Comment:
Will the user see this warning in the notebook?
If so, can we make it more user friendly, maybe something like: "There is no need to define an @EntryPoint() attribute in a notebook, it will be ignored." (or similar)
'workspace' might be confused with Azure workspace and I don't think the user cares about compilation.

Reply:
This warning only goes to the logger, which won't be visible in the notebook itself. I agree though, "workspace" isn't the right word to use here, since it's overloaded both in the Azure sense and also in the [IQ# sense](https://docs.microsoft.com/en-us/qsharp/api/iqsharp-magic/workspace). I also don't want to say only "notebook", because this also applies to compiling code snippets from string literals in Python. Perhaps "...when compiling Q# code defined in notebook cells or in string literals".